pkgs/tools/text/kdiff3/default.nix

```diff
@@ -1,35 +1,18 @@
 {
-  mkDerivation, lib, fetchgit, fetchpatch,
+  mkDerivation, lib, fetchurl, # fetchgit, fetchpatch,
   extra-cmake-modules, kdoctools, wrapGAppsHook,
   kcrash, kconfig, kinit, kparts
 }:
 
 mkDerivation rec {
-  name = "kdiff3-${version}";
-  version = "1.7.0-2017-02-19";
+  pname = "kdiff3";
+  version = "1.8";
 
-  src = fetchgit {
-    # gitlab is outdated
-    url = https://anongit.kde.org/scratch/thomasfischer/kdiff3.git;
-    sha256 = "0znlk9m844a6qsskbd898w4yk48dkg5bkqlkd5abvyrk1jipzyy8";
-    rev = "0d2ac328164e3cbe2db35875d3df3a86187ae84f";
+  src = fetchurl {
+    url = "https://download.kde.org/stable/${pname}/${pname}-${version}.tar.xz";
+    sha256 = "083pz5c1w7l9h4sb8zz8a763yph5sk3mxnhpdykz1rrggy9f8p54";
   };
 
-  setSourceRoot = ''sourceRoot="$(echo */kdiff3/)"'';
-
-  patches = [
```

Comment:
The steps needed to test whether this patch's functionality is included in the new release can be inferred by its source at https://gitlab.com/vcunat/kdiff3/commit/61061262161d90c015e8b53d5b3ba7df758d4bc0

Whereas the original patch just added stubs so kdiff3 wouldn't crash when given `--qall`/`--cs`/`--config`/`--confighelp`/`--L1`/`--L2`/`--L3`, the new one [is documented to actually support these options](https://github.com/KDE/kdiff3/blob/39d921959fe933007d79bb22b5409294420580f1/doc/en/index.docbook#L304-L317), confirming your conjecture that the patch is not only inapplicable but also unnecessary. :)

Reply:
**Perfect**! Thanks for the great investigation and sharing what you found.  A+ would be reviewed again. ;)